### ISSUE TITLE
Sort values in snippet and sort snippet in admin list

### ISSUE THREAD

**Issue by moreplavec** [wagtail/wagtail#5578]
Is is possible to do two things? If not, please, take it as feature request.

1. sort snippet values:
Is it possible to sort snippet values by name (or value) to make it easier to navigate in admin list and while using in select list? Now it's sorted by id and it's hard to find or make sure if value exists. Same problem is when selecting values from list (for example picking tag, category or param), maybe it doesn't exist or i haven't found it.

2. sort snippets list:
Is it possible to sort registered snippets? Again by name or value? I have about 15 registered snippets and now it's hard to find i want to work with.

version: django/wagtail latest

**Comment by gasman** [comment]
Hi @moreplavec,
To order a model by a specific field (or set of fields), you can use the `ordering` option in the model's `Meta` options: https://docs.djangoproject.com/en/2.2/topics/db/models/#meta-options

Setting a manual ordering within the admin isn't currently supported, so happy to accept that as a feature request.

**Comment by moreplavec** [comment]
Thanks, I was searching wagtail doc so hard i forgot it's Django too :)

**Comment by cybertschunk** [comment]
an option to order snippets manually in the backend would be awesome! It would allow backend users to easily influence in which order snippets are displayed on the frontend.

**Comment by thibaudcolas** [comment]
See #8609 for recent discussion on the roadmap for snippets. I don’t see much about sorting on there, it could be worth someone making the case on that thread.

**Comment by zerolab** [comment]
#8890 adds the ability to sort the listing. ([code ref](https://github.com/wagtail/wagtail/pull/8890/files#diff-e1bcd86f7d82181d10043d53e7cc8eda34f657fe28c7a75c3768b707ad000e10R181-R191))

@laymonage could pitch in more on this

**Comment by laymonage** [comment]
That PR adds the "Updated at" column (and live/draft "Status" column for snippets that extend `DraftStateMixin`). You can order the snippets by the new columns, but by default they respect the ordering specified in the model's `Meta.ordering`.

Technically, now you can also customize the listing view to add more columns and make them orderable, but right now the available API is too low-level and undocumented. We are planning to make it easier to do such customisations (somewhat similar to how modeladmin does it), but we still need to design the API first, and that won't be here until at least the next feature release (after Wagtail 4.0).

**Comment by laymonage** [comment]
This has been implemented in https://github.com/wagtail/wagtail/pull/9147, which is available since Wagtail 4.1.

You can define a `list_display` attribute on a `SnippetViewSet` subclass to refer to fields, attributes, or methods on the model. If it's a database field, you will be able to click on the column header to sort by that field.

For example, in a `wagtail_hooks.py`.

```py
from wagtail.snippets.models import register_snippet
from wagtail.snippets.views.snippets import SnippetViewSet


class PersonViewSet(SnippetViewSet):
    list_display = ["first_name", "last_name"]


register_snippet(Person, viewset=PersonViewSet)
```


See the following docs for more details:
- https://docs.wagtail.org/en/stable/topics/snippets.html#customising-snippets-admin-views
- https://docs.wagtail.org/en/stable/reference/viewsets.html#wagtail.snippets.views.snippets.SnippetViewSet.list_display

We are still improving the customisations of the admin views for snippets (and the docs for doing so). Head over to #10206 for more details. We'd love to hear your feedback!
